### PR TITLE
[Azure Search] Add option to skip version list initialization

### PR DESCRIPTION
**Pull request by loic-sharma**
A good chunk of time is spent by Db2AzureSearch to initialize the internal Version List resource (see [this description](https://github.com/NuGet/NuGet.Services.Metadata/blob/0f467803bea0596ffd91e5f859c5b0c6f2c42792/src/NuGet.Services.AzureSearch/VersionList/IVersionListDataClient.cs#L9-L16)). This resource is used by catalog2azuresearch to update the Azure Search index, thus, it is unnecessary if an advanced user just wants to create a search index without keeping it up-to-date. This change adds an config to disable version list initialization to speed up db2azuresearch. On DEV, this reduces the total execution time by ~8.4%. 

This change also moves the existing `ReplaceContainersAndIndexes` and `SkipPackagePrefixes` into a new `Development` configuration section (see [NuGetDeployment#1343](https://nuget.visualstudio.com/NuGetMicrosoft/_git/NuGetDeployment/pullrequest/1343?_a=overview)):

```json
{
  ...

  "Db2AzureSearch": {
    ...

    "Development": {
      "ReplaceContainersAndIndexes": true,
      "DisableVersionListWriters ": true,
      "SkipPackagePrefixes": [
        "e2e"
      ]
    }
  }
}
```

Test build: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=3531671
Test release (catalog2azuresearch): https://devdiv.visualstudio.com/DevDiv/_releaseProgress?_a=release-pipeline-progress&releaseId=606421

### CODE REVIEW

File: src/NuGet.Services.AzureSearch/VersionList/NullVersionListDataClient.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using NuGetGallery;
+
+namespace NuGet.Services.AzureSearch
+{
+    public class NullVersionListDataClient : IVersionListDataClient
```
Comment (loic-sharma) [inline src/NuGet.Services.AzureSearch/VersionList/NullVersionListDataClient.cs]: ⚠ Add tests

File: src/NuGet.Services.AzureSearch/DependencyInjectionExtensions.cs [hunk 2]
```diff
@@ -106,7 +106,24 @@ private static void RegisterAzureSearchStorageServices(ContainerBuilder containe
                 .Keyed<ICloudBlobClient>(key);
 
             containerBuilder
-                .Register<IVersionListDataClient>(c => new VersionListDataClient(
+                .Register<IVersionListDataClient>(c =>
+                {
+                    var options = c.ResolveOptional<IOptionsSnapshot<Db2AzureSearchConfiguration>>();
```
Comment (joelverhagen) [inline src/NuGet.Services.AzureSearch/DependencyInjectionExtensions.cs]: Hm.... this seems hard to discover to me. I'm not a big fan of toggling business logic in DI.

Can we just not call the VersionListDataClient when `DisableVersionListWriters` is true?

File: src/NuGet.Services.AzureSearch/AzureSearchJobDevelopmentConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace NuGet.Services.AzureSearch
+{
+    public class AzureSearchJobDevelopmentConfiguration
```
Comment (loic-sharma) [inline src/NuGet.Services.AzureSearch/AzureSearchJobDevelopmentConfiguration.cs]: I'm not super stoked by this name, given that config is only used by catalog2azuresearch and db2azuresearch. Please let me know if you have any suggestions here!

File: tests/NuGet.Services.AzureSearch.Tests/BatchPusherFacts.cs [hunk 4]
```diff
@@ -91,6 +91,26 @@ public async Task UpdatesVersionListIfAllActionsAreDone()
...
+                _config.AzureSearchBatchSize = 1;
+                _developmentConfig.DisableVersionListWriters = true;
+                _target.EnqueueIndexActions(IdA, _indexActions);
+
+                await _target.FinishAsync();
+
+                Assert.Equal(5, _hijackBatches.Count);
+                Assert.Equal(3, _searchBatches.Count);
+
+                _versionListDataClient.Verify(
```
Comment (joelverhagen) [inline tests/NuGet.Services.AzureSearch.Tests/BatchPusherFacts.cs]: 👍

File: src/NuGet.Services.AzureSearch/DependencyInjectionExtensions.cs (hunk 2 above)
Comment (loic-sharma) [reply src/NuGet.Services.AzureSearch/DependencyInjectionExtensions.cs]: Done

**Pull request merged**